Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Components/Objects/CreateSpeckleBlockInstance.cs

```diff
@@ -0,0 +1,214 @@
+    Params.Input[4].Optional = true;
+
+    // TODO: Add Color and Material inputs when supported
+  }
+
+  protected override void RegisterOutputParams(GH_OutputParamManager pManager)
+  {
+    pManager.AddParameter(
+      new SpeckleBlockInstanceParam(),
+      "Block Instance",
+      "BI",
+      "Speckle Block Instance",
+      GH_ParamAccess.item
+    );
+
+    pManager.AddParameter(
+      new SpeckleBlockDefinitionWrapperParam(),
+      "Definition",
+      "D",
+      "Block Definition of the instance",
+      GH_ParamAccess.item
+    );
+
+    pManager.AddGenericParameter("Transform", "T", "Transform of the Block Instance", GH_ParamAccess.item);
+
+    pManager.AddTextParameter("Name", "N", "Name of the Block Instance", GH_ParamAccess.item);
+
+    pManager.AddParameter(
+      new SpecklePropertyGroupParam(),
+      "Properties",
+      "P",
+      "Properties of the Block Instance",
+      GH_ParamAccess.item
+    );
+
+    // TODO: Add when supported
+    // pManager.AddColourParameter("Color", "C", "Color of the Block Instance", GH_ParamAccess.item);
+    // pManager.AddParameter(new SpeckleMaterialParam(), "Material", "M", "Material of the Block Instance", GH_ParamAccess.item);
+  }
+
+  protected override void SolveInstance(IGH_DataAccess da)
+  {
+    SpeckleBlockInstanceWrapperGoo? inputInstance = null;
+    da.GetData(0, ref inputInstance);
+
+    SpeckleBlockDefinitionWrapperGoo? inputDefinition = null;
+    da.GetData(1, ref inputDefinition);
+
+    IGH_Goo? inputTransform = null;
+    da.GetData(2, ref inputTransform);
+
+    string? inputName = null;
+    da.GetData(3, ref inputName);
+
+    IGH_Goo? inputProperties = null;
+    da.GetData(4, ref inputProperties);
+
+    // Create or copy result
+    SpeckleBlockInstanceWrapper result;
+    bool mutated = false;
+
+    if (inputInstance?.Value != null)
+    {
+      result = inputInstance.Value.DeepCopy();
+    }
+    else
+    {
+      result = new SpeckleBlockInstanceWrapper { Name = "Block Instance", ApplicationId = Guid.NewGuid().ToString() };
+      mutated = true;
+    }
+
+    // Process definition
+    if (inputDefinition?.Value != null)
+    {
+      result.Definition = inputDefinition.Value;
+      result.InstanceProxy.definitionId = inputDefinition.Value.ApplicationId ?? inputDefinition.Value.Name;
+      mutated = true;
+    }
+
+    // Process transform
+    if (inputTransform != null)
+    {
+      Transform? extractedTransform = ExtractTransform(inputTransform);
+      if (extractedTransform.HasValue)
+      {
+        result.Transform = extractedTransform.Value;
+        mutated = true;
+      }
+      else
+      {
+        AddRuntimeMessage(
+          GH_RuntimeMessageLevel.Warning,
+          "Transform input is not valid. Only Transforms and Planes are accepted."
+        );
+        return;
+      }
+    }
+
+    // Process name
+    if (inputName is { Length: > 0 }) // with !string.IsNullOrEmpty(inputName) compiler still complains at assignment
```

Comment:
unfamiliar with this syntax 👀
 wouldn't length still throw if inputName is null?

```
if (inputName is string validName && !string.IsEmptyOrWhitespace(validName))
{ // set here
}
```